test/with_dummyserver/test_socketlevel.py

```diff
@@ -102,7 +102,7 @@ def socket_handler(listener):
             sock.close()
 
         self._start_server(socket_handler)
-        with HTTPConnectionPool(self.host, self.port) as pool:
+        with HTTPSConnectionPool(self.host, self.port) as pool:
```

Comment:
Huh, good catch here. Maybe we should try `self.buf.count(self.host.encode()) >= 2` in the verification too? MIght be interesting to see the results.

Reply:
Shouldn't it be one in the HTTPS case too? It only worked before because the request was plaintext and the host header ws captured, but now it doesn't.

Reply:
Doh, right. Disregard me :)